### PR TITLE
SRO Bonus stats UI

### PR DESCRIPTION
**Pull request by frzyc**
## Describe your changes

Add initial UI to add and edit SRO bonus stats, need additional work to link to calc.

## Issue or discord link

- 

## Testing/validation

image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: libs/sr/db/src/Database/DataManagers/LoadoutDataManager.ts [hunk 1]
```diff
@@ -24,6 +25,7 @@ export interface Loadout {
 
   buildTcIds: string[]
   optConfigId: string
+  bonusStats: Array<{ tag: Tag; value: number }>
```
Comment (nguyentvan7) [inline libs/sr/db/src/Database/DataManagers/LoadoutDataManager.ts:28]: just curious, why are we using Array<> syntax over []?

File: libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  setValue,
+  onDelete,
+}: {
+  tag: Tag
+  setTag: (tag: Tag) => void
+  value: number
+  setValue: (value: number) => void
+  onDelete: () => void
+}) {
+  // TODO: best way to infer percentage from tag?
```
Comment (nguyentvan7) [inline libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx:154]: This is the correct way

File: libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  const isPercent = (tag.name || tag.q)?.endsWith('_')
+  return (
+    <CardThemed bgt="light">
+      <CardContent
+        sx={{ display: 'flex', gap: 1, justifyContent: 'space-around' }}
+      >
+        <SqBadge sx={{ m: 'auto' }}>{tag.q}</SqBadge>
+        {tag.q === 'dmg_' && (
+          <ElementTypeDropdown
```
Comment (nguyentvan7) [inline libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx:163]: technically this doesn't matter for HSR. I'm not aware of any character or mechanic that we would support in SRO that can do multi-element

Comment (nguyentvan7) [reply libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx:163]: damageType1 and damageType2 would be better candidates, if we want to be technical about it

File: libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx
Comment (nguyentvan7) [inline libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx:1]: shouldn't really be in 'tabs', but i assume this is just temporary anyways

**Review by nguyentvan7 — APPROVED** [review]

File: libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx (hunk 3 above)
Comment (frzyc) [reply libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx:163]: I think you misunderstand. this differentiates between `dmg_` and `${element}_dmg_`, this is not a multi select.

File: libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx
Comment (frzyc) [reply libs/sr/page-team/src/TeammateDisplay/Tabs/BonusStats.tsx:1]: yeah im waiting for design team to come up with some preliminary before going into tabs

File: libs/sr/db/src/Database/DataManagers/LoadoutDataManager.ts (hunk 1 above)
Comment (frzyc) [reply libs/sr/db/src/Database/DataManagers/LoadoutDataManager.ts:28]: I think `{ tag: Tag; value: number }[]` looks jank, imo

Comment (lantua) [reply libs/sr/db/src/Database/DataManagers/LoadoutDataManager.ts:28]: You can use `TagMapEntries<number>`

**Pull request merged**